### ISSUE TITLE
Upgrade to Netty 4.0

### ISSUE THREAD

**Issue by cravergara** [elastic/elasticsearch#3226]
I'm currently working on this. Writing the issue here so we could discuss.

Netty 4 is not yet stable, so I won't merge until everything is tested and Netty 4 has a stable release.

**Comment by jasontedor** [comment]
> Feel free to ping me if any questions come up

@normanmaurer As [Netty 3.x is deprecated](http://stackoverflow.com/a/30477520) but is still currently maintained, do you have a rough timeframe for when it will no longer be the case that it is maintained?

**Comment by philnate** [comment]
Any news on when this will be done? It's causing some trouble with newer nettys, which use the same artefact names, but different packages. Could at least netty be shaded like the other dependencies are?

**Comment by jasontedor** [comment]
> Any news on when this will be done?

Not in the near future.

> Could at least netty be shaded like the other dependencies are?

We don't shade dependencies after the 1.x series, and are not going to make a change like that for the 1.x series.

**Comment by normanmaurer** [comment]
Just to be clear netty 4.x and 3.x use different groupId's and different packages so it should be no problem to use both versions in the same app.

**Comment by philnate** [comment]
Yeah, I got lost in dependencies. Somehow some old org.jetty.netty got pulled in which didn't fit elasticsearchs needs. I think what got me confused is that mvnrepository tells  that ES 2.2 uses io.netty:netty (which in fact has org.jetty.netty packages) http://mvnrepository.com/artifact/org.elasticsearch/elasticsearch/2.2.0, this package would conflict with never nettys. I've pulled in io.netty:netty:3.x and io.netty:netty-all:4.y and it worked out.
